### PR TITLE
reg: take slope and intercept into account

### PR DESCRIPTION
**Pull request by rijobro**
upon reading an image, convert its array such that data=slope*data+intercept. Then set slope and intercept to 0 and 1 respectively.

### CODE REVIEW

File: src/Registration/cReg/NiftiImageData.cpp [hunk 1]
```diff
@@ -222,6 +226,12 @@ NiftiImageData<dataType> NiftiImageData<dataType>::operator*(const float& val) c
     return maths(val,mul);
 }
 
+template<class dataType>
+NiftiImageData<dataType> NiftiImageData<dataType>::operator/(const float& val) const
```
Comment (KrisThielemans) [inline src/Registration/cReg/NiftiImageData.cpp]:
```suggestion
NiftiImageData<dataType> NiftiImageData<dataType>::operator/(const float val) const
```
generally never use a ref for `const float/int` etc. it's unclear and would just slow it down.

Comment (KrisThielemans) [reply src/Registration/cReg/NiftiImageData.cpp]: I see it's done with `*` as well though. Maybe accept this one, and then do a separate PR for all of those (not urgent)

File: src/Registration/cReg/NiftiImageData.cpp [hunk 2]
```diff
@@ -61,7 +61,9 @@ template<class dataType>
 NiftiImageData<dataType>::NiftiImageData(const NiftiImageData<dataType>& to_copy)
 {
     copy_nifti_image(_nifti_image,to_copy._nifti_image);
-    set_up_data(to_copy._original_datatype);
+    this->_data = static_cast<float*>(_nifti_image->data);
```
Comment (KrisThielemans) [inline src/Registration/cReg/NiftiImageData.cpp]: I'll take your word for it (is it always `float`?)

**Comment by rijobro** [comment]
When reading an image, I convert to `float` (and convert back at the end). This means that the static cast from `void *` to `float` is safe.

Nifti images in general can be complex, but I only accept a subset of all nifti images, so this isn't a problem. Accepted data types: DT_BINARY, DT_INT8, DT_INT16, DT_INT32,  DT_FLOAT32, DT_FLOAT64, DT_UINT8, DT_UINT16, DT_UINT32, DT_INT64, DT_UINT64, DT_FLOAT128.

**Pull request merged**